src/credentials.rs

```diff
@@ -0,0 +1,105 @@
+use crate::errors::SdError;
+use nix::dir;
+use nix::fcntl::OFlag;
+use nix::sys::stat::Mode;
+use std::env;
+use std::fs::File;
+use std::path::PathBuf;
+
+/// Credential loader for units.
+///
+/// Credentials are read by systemd on unit startup and exported by their ID.
+///
+/// **Note**: only the user associated with the unit and the superuser may access credentials.
+///
+/// More documentation: <https://www.freedesktop.org/software/systemd/man/systemd.exec.html#Credentials>
+#[derive(Debug)]
+pub struct CredentialsLoader {
+    path: PathBuf,
+    _dirfd: dir::Dir,
+}
+
+impl CredentialsLoader {
+    /// Try to open credentials directory.
+    pub fn open() -> Result<Self, SdError> {
+        let path = Self::path_from_env().ok_or_else(|| {
+            SdError::from("No valid environment variable 'CREDENTIALS_DIRECTORY' found")
```

Comment:
I'd use IOError with NotFound here so that we can expose IOError from this function 🤔